config/config.go

```diff
@@ -15,138 +15,474 @@ package config
+	// inhibited by the source alerts).
+	TargetFilters []*Filter `yaml:"target_filters,omitempty"`
+
+	// A set of label names whose label values need to be identical in source and
+	// target alerts in order for the inhibition to take effect.
+	MatchOn []string `yaml:"match_on,omitempty"`
+
+	// How many seconds to wait for a corresponding inhibit source alert to
+	// appear before sending any notifications for active target alerts.
+	// TODO(julius): Not supported yet. Implement this!
+	// optional int32 before_allowance = 4 [default = 0];
+	// How many seconds to wait after a corresponding inhibit source alert
+	// disappears before sending any notifications for active target alerts.
+	// TODO(julius): Not supported yet. Implement this!
+	// optional int32 after_allowance = 5 [default = 0];
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (r *InhibitRule) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain InhibitRule
+	if err := unmarshal((*plain)(r)); err != nil {
+		return err
 	}
+	return checkOverflow(r.XXX, "inhibit rule")
+}
 
-	for _, a := range c.AggregationRule {
-		for _, f := range a.Filter {
-			if f.NameRe == nil {
-				return fmt.Errorf("Missing name pattern (name_re) in filter definition: %s", proto.MarshalTextString(f))
-			}
-			if f.ValueRe == nil {
-				return fmt.Errorf("Missing value pattern (value_re) in filter definition: %s", proto.MarshalTextString(f))
-			}
-		}
+// Grouping and notification setting definitions for alerts.
+type AggrRule struct {
+	// Filters that define which alerts are matched by this AggregationRule.
+	Filters []*Filter `yaml:"filters,omitempty"`
 
-		if _, ok := ncNames[a.GetNotificationConfigName()]; !ok {
-			return fmt.Errorf("No such notification config: %s", a.GetNotificationConfigName())
-		}
+	// How many seconds to wait before resending a notification for a specific alert.
+	RepeatRate model.Duration `yaml:"repeat_rate"`
+
+	// Notification configurations to use for this AggregationRule, referenced
+	// by their name.
+	NotificationConfigs []string `yaml:"notification_configs"`
+
+	// Notify when resolved.
+	SendResolved bool `yaml:"send_resolved"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (r *AggrRule) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*r = DefaultAggrRule
+	type plain AggrRule
+	if err := unmarshal((*plain)(r)); err != nil {
+		return err
 	}
+	if len(r.NotificationConfigs) == 0 {
+		return fmt.Errorf("aggregation rule needs at least one notification config")
+	}
+	return checkOverflow(r.XXX, "aggregation rule")
+}
 
-	return nil
+// A regex-based label filter used in aggregations.
+type Filter struct {
+	Name  string  `yaml:"name"`
+	Regex *Regexp `yaml:"regex"`
```

Comment:
This was represented in this way because protobufs didn't have maps. In YAML, it would be better to represent this as key-value pairs. Ok to do in a later change though...